src/core/polyhedral/scop.cc

```diff
@@ -179,25 +179,46 @@ void checkFiltersDisjointStatements(const ScheduleTree* root) {
 }
 } // namespace
 
-void Scop::promoteGroupToShared(
+void Scop::promoteGroup(
+    PromotedDecl::Kind kind,
     isl::id tensorId,
     std::unique_ptr<TensorReferenceGroup>&& gr,
     ScheduleTree* tree,
-    const std::unordered_set<isl::id, isl::IslIdIslHash>& activeStmts,
     isl::union_map schedule,
     bool forceLastExtentOdd) {
+  auto activePoints = activeDomainPoints(scheduleRoot(), tree);
+
+  for (const auto& kvp : activePromotions_) {
+    if (kvp.first.intersect(activePoints).is_empty()) {
+      continue;
+    }
+
+    auto groupId = kvp.second.groupId;
+    if (promotedDecls_.count(groupId) != 0 &&
+        promotedDecls_[groupId].tensorId == tensorId) {
+      // FIXME: allow double promotion if copies are inserted properly,
+      // in particular if the new promotion is strictly smaller in scope
+      // and size than the existing ones (otherwise we would need to find
+      // the all the existing ones and change their copy relations).
+      std::cerr << "FIXME: not promoting because another promotion of tensor "
```

Comment:
actually this spews non trivial amounts of text, how about ```LOG(WARNING)``` ?

Reply:
just kill it